### PR TITLE
Tests refactoring

### PR DESCRIPTION
**Pull request by Stranger6667**
Hello!
I'm doing some refactorings & bugfixes to the current test suite. It is not completed yet, but I'll try to do my best here :) These are more like proposed changes, because the current test suite is made with different testing styles, if something is not working, etc. So, I want to discuss the changes and improve django-money testing suite :)

#### Testing framework
Current master uses Django runner, which doesn't run any tests at all and py.test. So, I decided to completely switch to py.test. Not so many features are used at this point, but it seems useful to me.
Tox is still used for multi env testing, but with some simplifications and compatibility fixes. 

#### Compatibility and test builds
* Django master is supported
* Python 3.2 is supported completely
* PyPy3 builds were added. It works for Django 1.5/1.6/1.7/1.8
* Added test with django-money-rates and it works
* Django 1.5.x on Python 3.2 is fixed with proper django-reversion version (new 1.7.2 version is available on PyPI now).

Also coverage was added to test builds and I'll add codecov.io integration later.
Some builds were excluded from build matrix, based on Python versions support in Django. E.g. Django 1.9 doesn't support Python 3.2/3.3.

#### Found bugs and issues
I've found some bugs during refactoring.
1. Q objects generates different queries then regular kwargs passed to manager.filter() on Django 1.8+
It doesn't add `currency` part to WHERE clause. I've skipped equal parts (Python 2.7, Django 1.9)
```python
>>> print str(ModelWithTwoMoneyFields.objects.filter(Q(amount1__gt=F('amount2'))).query)
... WHERE "testapp_modelwithtwomoneyfields"."amount1" > ("testapp_modelwithtwomoneyfields"."amount2")
>>> print str(ModelWithTwoMoneyFields.objects.filter(amount1__gt=F('amount2')).query)
... WHERE ("testapp_modelwithtwomoneyfields"."amount1" > ("testapp_modelwithtwomoneyfields"."amount2") AND "testapp_modelwithtwomoneyfields"."amount1_currency" = ("testapp_modelwithtwomoneyfields"."amount2_currency"))
```
On Django versions prior to 1.8 queries are logically equal.
2. Multiplication & division on F objects. Addition and subtraction are implemented, but these operations are not.
Later I'll create separate issues for these things.

#### Cleanups
Shell scripts for configs generation were removed because they was very out of date and current Tox & Travis CI configs are simple enough to be handled manually.
All non-relevant extensions were removed from MANIFEST.in, also MANIFEST was removed as obsolete.
.gitignore was also updated to cover all required stuff (coverage reports, uncovered eggs).
six is not required as a dependency. The only usage of `six` is `django.utils.six`.
Code was slightly pep8ified.

#### New features
Universal wheel config were added, now PyPI package holder can do `python setup.py bdist_wheel` and upload the universal wheel.
Makefile. I've added that to simplify development. Now it is possible to clean up working dir, to see the coverage, to run the test suite with `make`.
Isort was applied. Just `isort -rc .` and imports will be organized in the same way in every file.

#### Questions
1. Why there are test builds against `py-moneyed==0.4`? Latest version as requirement for django-money is not enough?
2. About rounding. Consider this code:
```python
money = Money('100.0623456781123219')
instance = ModelWithVanillaMoneyField.objects.create(money=money)
# Now instance.money is not rounded
retrieved = ModelWithVanillaMoneyField.objects.get(pk=instance.pk)
# retrieved.money is rounded to 100.06
```
Should instance.money be rounded?

So, hope you'll find this PR interesting :) Discussion is welcome

### CODE REVIEW

**Comment by benjaoming** [comment]
Great work!! For the future of the project, it's really important to have solid tests.. and well yeah, it has to do with money so even more important to have discussions like the one about rounding.

> Why there are test builds against py-moneyed==0.4? Latest version as requirement for django-money is not enough?

Nah, I would believe it's a mistake. The latest releases (or master) of django-money can depend on the latest issues of any dependency as long as it passes tests.

> About rounding. Consider this code:

So the question is: If we store a value with high precision, but we can only ever retrieve it with lower precision, should we store it with this precision, too?

I would definitely say YES. Because if you are accounting money, you always go for rounded numbers. For instance, if currency convertion gives you lots of decimals, you throw them away immediately, otherwise further calculations will be untransparent.

Let's say all our reports and prints say:

1.00
1.00
1.00
1.00
1.00

The sum of that is 5.00

But what if we had this in the database:


1.001
1.001
1.001
1.001
1.001

The sum (using `aggregates.Sum`) of that is 5.005 and would display as 5.01. That result would be impossible to understand and would be very very questionable if used in accounting.

Would you change this to rounding pre-storage? I'm totally :+1: for that!

**Comment by Stranger6667** [comment]
Hmm, there are some fancy "Permission denied" errors for PyPy3 on Travis. I'll remove PyPy3 from .travis.yml

**Comment by Stranger6667** [comment]
About rounding. I'd rather create separate issue for that and continue work on rounding after this PR.
Thanks :)

File: MANIFEST.in [hunk 1]
```diff
@@ -2,5 +2,5 @@ include setup.py
 include CHANGES.txt
 include LICENSE.txt
 include README.rst
-include runtests.py
-recursive-include djmoney *.py *.html *.txt *.png *.js *.css *.gif *.less *.mo *.po *.otf *.svg *.woff *.eot *.ttf
+recursive-include djmoney *.py
+recursive-include tests *.py
```
Comment (benjaoming) [inline MANIFEST.in]: This `tests` should not be included as we are not going to install it on a user's system and having a global `tests` package would probably create conflict

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	rm -fr htmlcov/
+
+test:
+	python setup.py test --pytest-args="--cov=djmoney"
+
+coverage:
+	coverage run --source djmoney setup.py test
+	coverage report -m
+	coverage html
+	open htmlcov/index.html
```
Comment (benjaoming) [inline Makefile]: open doesn't work on non-osx systems, removing it is probably better

File: djmoney/_compat.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# coding: utf-8
```
Comment (benjaoming) [inline djmoney/_compat.py]: GREAT REFACTOR, thanks for isolating all this!

**Comment by benjaoming** [comment]
Is this still WIP? It looks great!

Feel free to create a new issue about the rounding stuff.

Out of curiosity: Is codecov-io better than coveralls.io? I'm afraid I've experienced a lot of interruptions with coverage.io, namely that it seems to have totally stopped working on some projects :/

**Comment by Stranger6667** [comment]
I'll apply your comments and then it will be ready to merge.
Unfortunately I can't tell much about codecov-io vs coveralls.io ... In my experience everything works fine with codecov-io. I tried coveralls year ago and found some issues with unified coverage reports. It just didn't work for me. Some time ago I tried codecov-io and everything was ok.
So it is more question of my personal preference here. If some problems are there, then we can try coveralls

**Comment by benjaoming** [comment]
@Stranger6667 could you try changing the .travis config to be container based? I think the slow build is due to this:

> Build delays for travis-ci.org (public) sudo enabled Precise and Trusty builds

https://www.traviscistatus.com/incidents/pfcclfpffppn

**Comment by Stranger6667** [comment]
Still not started :(

**Comment by benjaoming** [comment]
I think it was stuck waiting for the prev. build so I've cancelled that one...

**Comment by benjaoming** [comment]
Thanks for changing to cont. based!

**Comment by Stranger6667** [comment]
You are welcome :)

**Comment by Stranger6667** [comment]
Ooops :) Docker image for Python 3.5 is not available. I'll update .travis.yml

**Comment by Stranger6667** [comment]
Everything seems to be OK. PR is ready to be merged :)

**Comment by benjaoming** [comment]
Hurray! Great work!

**Pull request merged**